docs/content/users/basics/architecture.md

```diff
@@ -0,0 +1,156 @@
+: Contains examples and implementations to demonstrate how the [`ddev pull`](../basics/commands.md#pull) command can work with [hosting providers](../providers/index.md).
+
+`web-build` directory
+: Can be used to provide a [custom Dockerfile](../extend/customizing-images.md) for the web container.
+
+`xhprof` directory
+: Contains the `xhprof_prepend.php` file that can be used to customize [xhprof](../debugging-profiling/xhprof-profiling.md) behavior for different types of websites.
+
+#### Look But Don’t Touch
+
+Files beginning with `.` are hidden because they shouldn’t be fiddled with; most are regenerated, and thus overwritten, on every `ddev start`:
+
+`.dbimageBuild` directory
+: The generated Dockerfile used to customize the `db` container on first start.
+
+`.ddev-docker-compose-base.yaml`
+: The base docker-compose file used to describe a project.
+
+`.ddev-docker-compose-full.yaml`
+: The result of preprocessing `.ddev-docker-compose-base.yaml` using `docker-compose config`. Mostly it replaces environment variables with their values.
+
+`.gitignore`
+: The `.gitignore` is generated by DDEV and should generally not be edited or checked in. (It gitignores itself to make sure you don’t check it in.) It’s generated on every `ddev start` and will change as DDEV versions change, so if you check it in by accident it will always be showing changes that you don’t need to see in `git status`.
+
+`.global_commands`
+: Temporary directory used to get global commands available inside a project. You shouldn’t ever have to look there.
+
+`.homeadditions`
+: Temporary directory used to consolidate global `homeadditions` with project-level `homeadditions`. You shouldn’t ever have to look here.
+
+`.webimageBuild` directory
+: The generated Dockerfile used to customize the web container on first start.
+
+### Global Files
+
+There’s only one global `.ddev` directory, which typically lives in your home folder: `~/.ddev` or `$HOME/.ddev`.
+
+`global_config.yaml`
+: This YAML file defines your global configuration, which consists of various [config settings](../configuration/config.md) along with an important `project_info` key that lets DDEV keep track of the projects you’ve added.
+
+`bin` directory
+: This is where DDEV stores executable binaries it needs, like `mutagen` and `docker-compose`.
+
+`commands` directory
+: Directory for storing DDEV commands that should be available in containers, like `npm`, `artisan`, and `drush` for example. These are organized in subdirectories named for where they’ll be used: `db`, `host`, and `web`. You can add your own [custom commands](../extend/custom-commands.md) here.
+
+`homeadditions` directory
+: Just like the per-project `homeadditions` directory, files you add here will automatically be copied into the web container’s home directory. In this case of this _global_ homeadditions directory, the files will be copied into **every** web container’s home directory.
+
+#### Look But Don’t Touch
```

Comment:
```suggestion
#### Look But Don’t Touch (global config)
```

This fixes the docs test complaint. Can be fixed any other way you want.

Reply:
I guess I’d switch these to “Hidden Project Files” and “Hidden Global Files”.